### PR TITLE
Add r-value return from AES encrypt, insert H(P_spender) into encrypted payload

### PR DESCRIPTION
**Pull request by dbmcclain**
initial commit

### CODE REVIEW

**Comment by dbmcclain** [comment]
AES Encrypt now returns encrypted payload struct, plus the secret r-value used for keying
Hash of spender pubkey now stored in encrypted payload
Generating PaymentOutput now requires pubkey of spender, will be hashed and stored in encrypted payload.
New PaymentOutput functions return the UTXO, the gamma, and the r-value

**Comment by dbmcclain** [comment]
Added function to PaymentOutput:
verify_proof_of_payment(&UTXO, &spender_pkey, &r-value) -> Result<amount: i64, Error>

**Comment by dbmcclain** [comment]
fix mistaken keying on encrypted payload:
key = H(r * s *G) = H(r * P)
for P = uncloaked recipient key
so verify_proof_of_payment() needs UTXO, spender_pkey, r-value, uncloaked_recipient_pkey.

File: blockchain/src/genesis.rs [hunk 1]
```diff
@@ -59,6 +59,9 @@ pub fn genesis(
         // Genesis has one PaymentOutput + N * StakeOutput, where N is the number of validators.
         //
 
+        // TODO: use correct spender pubkey
```
Comment (rtsisyk) [inline blockchain/src/genesis.rs]: In genesis node[0] sends money to itself. There is no "correct" spender pubkey.

**Comment by dbmcclain** [comment]
> I have a conceptual question here - how storing of sender's public key in EncryptedPayload can prove that this UTXO was actually created by that wallet?

yes you are correct - we need a signature instead of key-hash

**Comment by dbmcclain** [comment]
just sync'd with dev branch

**Comment by rtsisyk** [comment]
I will review it after v0.8 release.

**Comment by rtsisyk** [comment]
Superseded by https://github.com/stegos/stegos/pull/970

**Pull request closed without merging**